polus-graph-pyramid-builder-plugin/src/main.py

```diff
@@ -131,46 +491,135 @@ def bin_data(data,column_names):
             if feat2_tf.size<=1:
                 continue
             
+            
             # sort linear matrix indices
             feat2_sort = np.sort(feat1_tf[data_ind[name1] & data_ind[name2]] + feat2_tf)
             
             # Do math to get the indices
             ind2 = np.diff(feat2_sort)                       
             ind2 = np.nonzero(ind2)[0]                       # nonzeros are cumulative sum of all bin values
             ind2 = np.append(ind2,feat2_sort.size-1)
-            # print(feat2_sort.shape)
-            rows = (feat2_sort[ind2]/200).astype(np.uint8)   # calculate row from linear index
-            cols = np.mod(feat2_sort[ind2],200)              # calculate column from linear index
+            rows = (feat2_sort[ind2]/bincount).astype(np.uint8)   # calculate row from linear index
+            cols = np.mod(feat2_sort[ind2],bincount)              # calculate column from linear index
             counts = np.diff(ind2)                           # calculate the number of values in each bin
-            bins[feat1,feat2,rows[0],cols[0]] = ind2[0] + 1
-            bins[feat1,feat2,rows[1:],cols[1:]] = counts
+            bins[i,rows[0],cols[0]] = ind2[0] + 1
+            bins[i,rows[1:],cols[1:]] = counts
             linear_index.append([feat1,feat2])
-    return bins, bin_stats, linear_index
+            i = i + 1
+            
+    return yaxis, bins, bin_stats, linear_index, linear_dict, column_bin_size, alphavals
 
 """ 2. Plot Generation """
+def format_ticks_log(fmin,fmax,nticks, yaxis, commonratio, alphavalue):
+    """ Generate tick labels
+    Polus Plots uses D3 to generate the plots. This function tries to mimic
+    the formatting of tick labels. Tick labels have a fixed width, and in
+    place of using scientific notation a scale prefix is appen ded to the end
+    of the number. See _prefix comments to see the suffixes that are used.
+    Numbers that are larger or smaller than 10**24 or 10**-24 respectively
+    are not handled and may throw an error. Values outside of this range
+    do not currently have an agreed upon prefix in the measurement science
+    community.
+    Inputs:
+        fmin - the minimum tick value
+        fmax - the maximum tick value
+        nticks - the number of ticks
+    Outputs:
+        fticks - a list of strings containing formatted tick labels
+    """
```

Comment:
As with the comment for the binning method for log scaled data, it might be better to just modify the `format_ticks` functions to accept additional inputs rather than creating an entirely new function. This will making modifying the code in the future much simpler, since there is a lot of overlap in what the two functions do.